importer-db-consistency/src/Pos/ImporterDBConsistency/ConsistencyChecker.hs

```diff
@@ -0,0 +1,127 @@
+module Pos.ImporterDBConsistency.ConsistencyChecker
+  (
+    externalConsistencyFromBlk
+  , externalConsistencyRandom
+  , internalConsistencyCheck
+  , externalConsistencyWithTxRange
+  ) where
+
+import           Universum
+
+import           Data.List (tail)
+import           System.Wlog (logInfo)
+
+import qualified Pos.BlockchainImporter.Tables.TxsTable as TxsT (TxRecord)
+import           Pos.Core (BlockCount (..), HasPrevBlock (prevBlockL), HasProtocolConstants,
+                           HeaderHash, blkSecurityParam, headerHash)
+import           Pos.DB (getHeader, getTipHeader)
+import           Pos.GState.BlockExtra (resolveForwardLink)
+import           Pos.ImporterDBConsistency.Properties
+import           Pos.ImporterDBConsistency.Utils
+import           Pos.Txp (Tx)
+
+----------------------------------------------------------------------------
+-- Checks
+----------------------------------------------------------------------------
+
+{-
+  Check consistency with the key-value db of a node up-to-date.
+  Objective: Test consistency of the postgresdb generated after running full import,
+             with the one generated in a Cardano node.
+  Requires: Having the postgresdb also up-to-date
+    - Checks that the same best block is in both the node and postgresdb
+    - Checks that utxo from node are stored in postgresdb
+    - Checks that tx in node from all blocks starting from a given one are stored postgresdb
+-}
+externalConsistencyFromBlk :: ConsistencyCheckerEnv m => HeaderHash -> m Bool
```

Comment:
Maybe is a good idea to comment that this is a partial/total (if header hash corresponds to block 0) check of consistency between cardano node and the importer. And also that the "test" below (`externalConsistencyRandom`) is a statistical variation of this check.